src/Development/IDE/LSP/CodeAction.hs

```diff
@@ -53,14 +54,16 @@ codeLens
     -> CodeLensParams
     -> IO (List CodeLens)
 codeLens _lsp ideState CodeLensParams{_textDocument=TextDocumentIdentifier uri} = do
-    diag <- getDiagnostics ideState
+    -- diag <- getDiagnostics ideState
     case uriToFilePath' uri of
       Just (toNormalizedFilePath -> filePath) -> do
+        _ <- runAction ideState $ use_ TypeCheck filePath
```

Comment:
`use_` throws a `BadDependency` exception if the file does not typecheck. You probably want `useE` which will give you a `Maybe`.

Reply:
Resolved in a new commit.